src/abstractdataframe/io.jl

```diff
@@ -1,17 +1,90 @@
+"""
+    show(io::IO, mime::MIME, df::AbstractDataFrame;
+         allrows::Bool = !get(io, :limit, false),
+         allcols::Bool = !get(io, :limit, false),
+         splitcols = get(io, :limit, false),
+         rowlabel::Symbol = :Row,
+         summary::Bool=true)
+
+Render a data frame to an I/O stream in MIME type `mime`.
+
+# Positional arguments
+- `io::IO`: The I/O stream to which `df` will be printed.
+- `mime::MIME`: supported MIME types are: `"text/plain"`, `"text/html"`, `"text/latex"`,
+  `"text/csv"`, `"text/tab-separated-values"`
+- `df::AbstractDataFrame`: The data frame to print.
+
+# Keyword arguments
+- MIME type `"text/plain"` accepts all listed keyword arguments and therir behavior
+  is identical as for `show(::IO, ::AbstractDataFrame)`
+- MIME types `"text/html"`, `"text/latex"` accept only `allrows` and `allcols`
+  keyword arguments. They allow the user to decide to print all rows/columns
+  instead of the default values given in `HTML_NROWS`, `LATEX_NROWS`,
+  `HTML_NCOLS`, and `LATEX_NCOLS` constants.
+  Additionally `"text/html"` MIME type accepts `summary` keyword argument which
+  allows to choose whether to print a brief string summary of the data frame.
+- MIME types `"text/csv"`, `"text/tab-separated-values"` do not support any
+  keyword arguments.
+
+# Examples
+```jldoctest
+julia> show(stdout, MIME("text/latex"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+\\begin{tabular}{r|cc}
+        & A & B\\\\
+        \\hline
+        & Int64 & String\\\\
+        \\hline
+        1 & 1 & x \\\\
+        2 & 2 & y \\\\
+        3 & 3 & z \\\\
+\\end{tabular}
+14
+
+julia> show(stdout, MIME("text/csv"), DataFrame(A = 1:3, B = ["x", "y", "z"]))
+"A","B"
+1,"x"
+2,"y"
+3,"z"
+```
+"""
+Base.show(io::IO, mime::MIME, df::AbstractDataFrame;
+          allrows::Bool = !get(io, :limit, false),
+          allcols::Bool = !get(io, :limit, false),
```

Comment:
Are these keywords really necessary?   The user could always pass `IOContext(io, :limit=>false)`, or `IOContext(io, :displaysize=>(10^9,10^9))` if you honor `displaysize…

Reply:
The idea was to allow to independently change this for rows and columns. But given the discussion in this PR it seems that the whole approach is flawed. Thank you for the comment.

Currently on master we have the following rule:
```
    haslimit = get(io, :limit, true)
    if haslimit
        tty_rows, tty_cols = displaysize(io)
        mxrow = min(nrows,tty_rows)
    else
        mxrow = nrows
    end
```
so maybe we leave it and just decide on number of columns to print based on the same rules as we use for REPL vs `tty_cols`?

The only question is if we should base these rules on `displaysize` function (this is currently on master) or `:displaysize` property from `IOContext` (and fallback to `displaysize` function when it is not defined)?

Reply:
You should base it on the displaysize property from IOContext.

Via displaysize you could independently remove the limit for the rows and columns just by setting the rows or columns to a large number.

Reply:
We added the keywords for convenience in interactive use at the REPL. I'd say better support them for all methods, even if the same behavior can be achieved using `IOContext`.

I agree it sounds good to use the same rules as for the REPL. But note that calling `displaysize(io)` or getting `:displaysize` from `IOContext` is equivalent AFAICT, as the former uses the latter when the stream is an `IOContext`. Looking up the `IOContext` property directly just allows you to use a default different from `displaysize()` if you want.

Reply:
I am OK with any solution (to be frank I would not expect the functionality to be used much in practice anyway and I understand we are going to rewrite what we have here in the future anyway; we can keep in mind that currently `allrows` and `allcols` kwargs are not supported, so adding them will add another deprecation period when we remove them), so please decide and I can implement whatever is the concensus.